### PR TITLE
Fix bug in gunzip udf

### LINKED ISSUE #369: Gunzip Error
Error Msg: ` TypeError: CreateListFromArrayLike called on non-object at udf_js_gunzip(BYTES) line 4, columns 35-36 `

Query:
```

CREATE TEMP FUNCTION
  udf_js_gunzip (input BYTES)
  RETURNS STRING
  LANGUAGE js AS """
    // converts a byte array to a string
    function binary2String(byteArray) {
        return String.fromCharCode.apply(String, byteArray);
    }
    
    // input is in base64, so it needs to be decoded
    var decodedData = atob(input);
    var compressedData = decodedData.split('').map(function(e) {
        return e.charCodeAt(0);
    });
    
    try {
      var gunzip = new Zlib.Gunzip(compressedData); 
    
      // decompress returns bytes that need to be converted into a string
      var unzipped = gunzip.decompress();
      return binary2String(unzipped);
    } catch (err) {
      return binary2String(input);
    }
"""
OPTIONS (
  library = "gs://moz-fx-data-circleci-tests-bigquery-etl/gunzip.min.js",
  library = "gs://moz-fx-data-circleci-tests-bigquery-etl/atob.js"
);

WITH ping_counts AS (
  SELECT
    metadata.document_namespace,
    metadata.document_type,
    metadata.document_version,
    COUNT(*) AS ping_count
  FROM
    `payload_bytes_decoded.structured_*`
  WHERE
    DATE(submission_timestamp) >= DATE_SUB(current_date, INTERVAL 1 day)
  GROUP BY ROLLUP(document_namespace, document_type, document_version)
), error_counts AS (
  SELECT
    document_namespace,
    document_type,
    document_version,
--    error_message,
--  udf_js_gunzip(ARRAY_AGG(payload)[OFFSET(0)]) AS example_payload,
    COUNT(*) AS error_count
  FROM
    payload_bytes_error.structured
  WHERE
    error_type = "ParsePayload"
    AND DATE(submission_timestamp) >= DATE_SUB(current_date, INTERVAL 1 day)
  GROUP BY ROLLUP(document_namespace, document_type, document_version)
), example_errors AS (
  SELECT
    document_namespace,
    document_type,
    document_version,
    error_message,
    udf_js_gunzip(ARRAY_AGG(payload)[OFFSET(0)]) AS example_payload
  FROM
    payload_bytes_error.structured
  WHERE
    error_type = "ParsePayload"
    AND DATE(submission_timestamp) >= DATE_SUB(current_date, INTERVAL 1 day)
  GROUP BY
    document_namespace,
    document_type,
    document_version,
    error_message
)

SELECT
  document_namespace,
  document_type,
  document_version,
  ping_count,
  error_count,
  1.0 * error_count / ping_count AS error_ratio,
  error_message,
  example_payload
FROM
  ping_counts
INNER JOIN
  error_counts USING (document_namespace, document_type, document_version)
INNER JOIN
  example_errors USING (document_namespace, document_type, document_version)
ORDER BY error_ratio DESC
```

### PR DESCRIPTION
**Pull request by fbertsch**
Fixes #369

### CODE REVIEW

**Comment by fbertsch** [comment]
There are a lot of encodings going on here so I attempted to document them a bit more.

File: udf_js/gunzip.sql [hunk 1]
```diff
@@ -12,13 +12,24 @@ CREATE TEMP FUNCTION
...
+     *    - A gzipped UTF-8 byte array
+     *    - A UTF-8 byte array
+     *
+     *  Outputs a string representation
+     *  of the byte array (gunzipped if
+     *  possible).
+     */
+
     function binary2String(byteArray) {
+        // converts a UTF-16 byte array to a string
```
Comment (relud) [inline udf_js/gunzip.sql]: the comment before this function suggests that we should be decoded a UTF-**8** byte array to a string?
```suggestion
        // converts a UTF-8 byte array to a string
```

Comment (fbertsch) [reply udf_js/gunzip.sql]: [All javascript strings are internally UTF-16,](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/fromCharCode) so this is unfortunately all a bit confusing to run

Comment (fbertsch) [reply udf_js/gunzip.sql]: I see where the confusion lies though, so I fixed the below documentation

**Pull request merged**